pkg/controller/direct/bigquerydataset/dataset_controller.go

```diff
@@ -184,6 +184,20 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 
 	resource := clone.Clone(a.actual).(*api.Dataset)
 
+	// If the location is not specified upon creation, merge the desired with the previously applied location defaulting.
+	presviouslyApplied, _, _ := unstructured.NestedMap(u.Object, "status")
```

Comment:
nit: does `unstructured.NestedString(u.Object, "status", "observedState", "location")` not working?